std/container/array.d

```diff
@@ -539,11 +525,13 @@ Complexity: $(BIGOH 1)
     }
 
 /**
-Ensures sufficient capacity to accommodate $(D e) elements.
+Ensures sufficient capacity to accommodate $(D e) _elements.
+If $(D e < capacity), this method will not decrease the capacity.
 
 Postcondition: $(D capacity >= e)
 
-Complexity: $(BIGOH 1)
+Complexity: $(BIGOH e) if $(D e > capacity),
```

Comment:
Does the amortized complexity apply in this case (honest question)? What about this clearly stupid code:
```d
Array!int a;
foreach (i; 0 .. 1000) {
   a.reserve(a.length + 1); // assume realloc every time copy all elements
   a.insertBack(i);
}
```